### ISSUE TITLE
FormData helper function

### ISSUE THREAD

**Issue by dreitzner** [sveltejs/kit#9715]
### Describe the problem

We have currently no connection between forms and form actions. (I know this is a tricky topic)
It might be cool to improve the DX here, because forms are such an important part of the resilient application story of SvelteKit (works with and without JS)

If I am not mistaken, we should be able to extract some information on the route level.

## This would enable the following patterns:

- extract (typed) objects from FormData with a helper function
- possibly guard against parsing errors
- show warning in `.svelte` files if forms are lacking required input fields 
- show warnings, if form action is not available (e.g. `'/upsi' does not have form action 'help' `
- possibly a thousand other cool things I can't think of right now

## Things to think about:

- files should probably be extracted separately
- it might be good to have the possibility to mark fields as optional with something like `sveltekit-form-input-optional` or make use of `required` 
- this would only work in routes where there is a form present in the `+.page.svelte` file and the 

### Describe the proposed solution

Basic example of a helper function:
```svelte
<!-- /test/+page.svelte -->
<form method="POST" enctype="multipart/form-data">
  <input type="text" name="text" id="text">
  <button type="submit">submit big file</button>
</form>
```
```js
// /test/+page.server.js
export const actions = {
    default: async ({request}) => {
        const data = await proccesFormdata(request)
       // have a nicly processed data dependent on the request
       // do easy validation, for example with zod
    }
};
```

I know that form actions can be called from different routes, but maybe we can check that at compile time as well?
The example below could maybe show a warning like `Form Action for '/test' also expects field with name="text"`
```svelte
<!-- /other/+page.svelte -->
<form action="/test" method="POST" enctype="multipart/form-data">
  <input type="text" name="upsi" id="text">
  <button type="submit">submit big file</button>
</form>
```

### Importance

would make my life easier

**Comment by dreitzner** [comment]
@dummdidumm ok, I agree with your viewpoint.

I'll close it in favor of leaving that part up to third parties.
Maybe we'll get some sort of form-renderer in superforms, that takes in a zod schema 😊

**Comment by ciscoheat** [comment]
@dreitzner the Zod schema can be introspected to a great degree, I'm using it to create constraints for the fields based on min/max values, etc, so you could use it to generate input fields based on a schema on the client. 

I had that thought for a while, there is even some simple metadata available, but I'll scrap that for 1.0 and will let the Zod schema be the metadata. That should make it possible to build a very flexible form-renderer on top of Superforms, or even without it.
